AwsCryptographicMaterialProviders/dafny/AwsCryptographicMaterialProviders/test/Keyrings/AwsKms/AwsKmsHierarchicalKeyring/TestAwsKmsHierarchicalKeyring.dfy

```diff
@@ -320,6 +324,477 @@ module TestAwsKmsHierarchicalKeyring {
         == decryptionMaterialsOut.materials.plaintextDataKey;
   }
 
+  method {:test} TestSharedCacheWithSamePartitionIdAndSameLogicalKeyStoreName()
```

Comment:
all these tests have a lot of shared boiler plate code, pls refactor into common methods similar to GetTestMaterials()

Reply:
I agree that these look very similar. But if you look closely, they're actually tweaking the logicalKeyStoreName and partitionId in all of the tests. There isn't much scope for refactoring.

Reply:
Unfortunately, as much as I'd want to refactor the tests, setting the Partition ID and Logical Key Store Name in every keystore / keyring initialization doesn't allow room for refactoring.